lua/swenv/api.lua

```diff
@@ -6,13 +6,34 @@ local ORIGINAL_PATH = vim.fn.getenv('PATH')
 
 local current_venv = nil
 
+local get_conda_base = function(conda_exe_path)
+  local parts = {}
+
+  for part in conda_exe_path:gmatch("[^/]+") do
+    table.insert(parts, part)
+  end
+  table.remove(parts) -- remove the last part
+  table.remove(parts) -- remove the second-to-last part
+  return "/" .. table.concat(parts, "/")
+end
+
 local set_venv = function(venv)
-  current_venv = venv
-  local venv_bin_path = venv.path .. '/bin'
-  vim.fn.setenv('PATH', venv_bin_path .. ':' .. ORIGINAL_PATH)
-  vim.fn.setenv('VIRTUAL_ENV', venv.path)
-  if settings.post_set_venv then
-    settings.post_set_venv(venv)
+  if venv.source == "conda" then
+    vim.fn.setenv('CONDA_PREFIX', venv.path)
+    vim.fn.setenv('CONDA_DEFAULT_ENV', venv.name)
+    vim.fn.setenv('CONDA_PROMPT_MODIFIER', "(" .. venv.name .. ")")
+    vim.cmd("LspRestart")
+    -- TODO: remove old path
+    local conda_base_path = get_conda_base(vim.fn.getenv("CONDA_EXE"))
+    vim.fn.setenv('PATH', venv.path .. '/bin' .. ':' .. ORIGINAL_PATH)
```

Comment:
maybe we could move this to a common function also used below, eg `update_path` or similar?